File: gitlab-nic/code-samples-main/API/Python/create-domain.py
#!/usr/bin/env python3

import requests
import json
import sys
import afnic

if len(sys.argv) != 2:
    raise Exception("Usage: create-domain name")
contact = "CTC65093"
args = {'name': sys.argv[1], "authorizationInformation": "Vachement1234sur",
        "registrantClientId": contact, "contacts": [
            {
                "clientId": contact,
                "role": "ADMINISTRATIVE"
            },
            {
                "clientId": contact,
                "role": "TECHNICAL"
            }
        ]} 
headers = afnic.get_headers()
response = requests.post("https://api-sandbox.nic.fr/v1/domains",
                         headers=headers, data=json.dumps(args))
try:
    body = json.loads(response.text)
except json.JSONDecodeError: # For some errors (5xx), the API returns HTML :-(
    body = response.text
print(body)
if response.status_code != 201:
    raise Exception("HTTP status code for create is %s" % response.status_code)
    

File: gitlab-nic/code-samples-main/API/Elixir/create-domain.exs
url = "https://api-sandbox.nic.fr/v1/domains"
contact = "CTC65093"

if length(System.argv()) != 1 do
  raise "Boom Usage is only one parameter, the domain name"
end

domain = List.first(System.argv())

body =
  Jason.encode!(%{
    name: domain,
    authorizationInformation: "Vachement1234sur",
    registrantClientId: contact,
    contacts: [
      %{
        clientId: contact,
        role: "ADMINISTRATIVE"
      },
      %{
        clientId: contact,
        role: "TECHNICAL"
      }
    ]
  })

result = HTTPoison.post!(url, body, Afnic.get_headers(), [])

response =
  case result do
    %HTTPoison.Response{status_code: 201, body: body} ->
      Jason.decode!(body)

    other ->
      raise "Boom (#{inspect(other)})"
  end

IO.puts("#{domain} created on #{response["creationDate"]}")
